src/SurveyService.Data/QuestionRepository.cs

```diff
@@ -17,15 +17,40 @@ public class QuestionRepository : IQuestionRepository
     _provider = provider;
   }
 
+  public async Task<Guid?> CreateAsync(DbQuestion dbQuestion)
+  {
+    if (dbQuestion is null)
+    {
+      return null;
+    }
+
+    _provider.Questions.Add(dbQuestion);
+    await _provider.SaveAsync();
+
+    return dbQuestion.Id;
+  }
+
   public Task<DbQuestion> GetAsync(Guid questionId)
   {
-    return _provider.Questions
-      .FirstOrDefaultAsync(x => x.Id == questionId);
+    return _provider.Questions.FirstOrDefaultAsync(x => x.Id == questionId);
+  }
+
+  public async Task<bool> CheckGroupProperties(Guid groupId, DateTime? deadline, bool hasRealTimeResult)
+  {
+    DbQuestion question = new DbQuestion();
+
+    question = await _provider.Questions.FirstOrDefaultAsync(q => q.GroupId == groupId);
+
+    if((question is null) || (question.Deadline != deadline) || (question.HasRealTimeResult != hasRealTimeResult))
```

Comment:
Maria said that it's okay to put part of the validation in the repository instead of making two queries into the database. We couldn't think of another solution.

Reply:
it is possible here to do one request to DB, receive dbQuestion and use it in validator